tests/testthat/test-public_api.R

```diff
@@ -21,3 +21,39 @@ test_that("styler does not return error when there is no file to style", {
 })
 
 
+##  ............................................................................
+
+dir <- tempfile("styler")
+dir.create(dir)
+region_path_temp <- paste0(dir, "/addin_region-in.R")
+region_path_perm <- paste0(base, "/xyzaddin/addin_region-in.R")
+file.copy(region_path_perm, dir)
+
+context <- structure(list(
+  id = "C533793B",
+  path = region_path_temp,
+  contents = c("fjkdsfa 2jy+wj/ 1+1 <?+d", ""),
+  selection = structure(list(structure(list(range = structure(list(
+    start = structure(c(1, 17), .Names = c("row", "column"), class = "document_position"),
+    end = structure(c(1, 21), .Names = c("row", "column"),
+                    class = "document_position")), .Names = c("start", "end"),
+    class = "document_range"), text = "1+1 "), .Names = c("range", "text"))),
+    .Names = "", class = "document_selection")),
+  .Names = c("id", "path", "contents", "selection"), class = "document_context"
+)
+
+# style_active_file() must be tested manually.
+test_that("styling active region works", {
+  reference <- utf8::read_lines_enc( paste0(base, "/xyzaddin/addin_region-out.R"))
```

Comment:
`file.path()` better explains your intentions. Even better, a small function

```r
testthat_file <- function(...) {
  file.path(rprojroot::find_testthat_root_file("public-api"), ...)
}
```

instead of the `base` variable. Here and elsewhere.